serverless/functions/reference-content/local-testing.mdx

```diff
@@ -0,0 +1,47 @@
+---
+meta:
+  title: Functions local testing
+  description: This section contains information about Serverless Function local development and testing.
+content:
+  h1: Functions runtimes configuration
+  paragraph: This section contains information about Serverless Function local development and testing
+tags: functions serverless local testing
+dates:
+  validation: 2023-03-06
+  posted: 2023-03-06
+categories:
+  - serverless
+---
+
+Local testing allows you to run your code on your dev environment with really short deploy time.
+
+Available frameworks :
+
+- [Go](https://github.com/scaleway/serverless-functions-go)
+- [Python](https://github.com/scaleway/serverless-functions-python)
+
+# Local testing
+
+Scaleway Serverless Functions can run outside Scaleway infrastructure, this is useful to setup,
+develop and test functions locally to be able to run it directly and analyse logs.
+
+Advantage of testing locally is to use your favorite debugger and test tools for your handler
+directly to analyse, debug, monitor and benchmark your function.
+
+## Requirements
+
+Local testing your function will create a local web server that listens to a given port.
+
+You will need to be able to install packages with the dependency manager of the runtime you use 
+(ex: `pip`, `go get`...).
+
+## Limits
+
+Building a function locally does not guarantee your function will work on Scaleway Serverless Functions.
```

Comment:
```suggestion
Each offline testing library does its best to replicate the Scaleway Functions environment, but cannot fully emulate the surrounding infrastructure. Therefore, although being able to run your function locally is a good indicator that it will work, it may still fail due to the execution environment used in Scaleway Serverless Functions.
```